scripts/sanitycheck

```diff
@@ -828,6 +834,13 @@ def main():
             else:
                 logger.error("""When --device-testing is used with --device-serial, only one
                       platform is allowed""")
+        elif options.device_serial_pty:
```

Comment:
this can be folded in the previous elif avoiding lots of duplication. load_device_from_cmdline could in this case take one serial option based on the arguments with the last option being boolean for example indicating it is a pty device. We also want to avoid the case where both --device-serial and --device-serial-pty are given, so the option should be conflicting in argparse I think.

Reply:
Fixed duplication issue, added mutually exclusive group serial for those parameters.